client/rest-high-level/src/main/java/org/elasticsearch/client/indexlifecycle/PhaseExecutionInfo.java

```diff
@@ -0,0 +1,135 @@
+        PARSER.declareObject(ConstructingObjectParser.constructorArg(), Phase::parse, PHASE_DEFINITION_FIELD);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), VERSION_FIELD);
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), MODIFIED_DATE_FIELD);
+    }
+
+    public static PhaseExecutionInfo parse(XContentParser parser, String name) {
+        return PARSER.apply(parser, name);
+    }
+
+    private final String policyName;
+    private final Phase phase;
+    private final long version;
+    private final long modifiedDate;
+
+    /**
+     * This class holds information about the current phase that is being executed
+     * @param policyName the name of the policy being executed, this may not be the current policy assigned to an index
+     * @param phase the current phase definition executed
+     * @param version the version of the <code>policyName</code> being executed
+     * @param modifiedDate the time the executing version of the phase was modified
+     */
+    public PhaseExecutionInfo(String policyName, Phase phase, long version, long modifiedDate) {
+        this.policyName = policyName;
+        this.phase = phase;
+        this.version = version;
+        this.modifiedDate = modifiedDate;
+    }
+
+    public String getPolicyName() {
+        return policyName;
+    }
+
+    public Phase getPhase() {
+        return phase;
+    }
+
+    public long getVersion() {
+        return version;
+    }
+
+    public long getModifiedDate() {
+        return modifiedDate;
+    }
+
+    public String getModifiedDateString() {
+        ZonedDateTime modifiedDateTime = ZonedDateTime.ofInstant(Instant.ofEpochMilli(modifiedDate), ZoneOffset.UTC);
+        return modifiedDateTime.toString();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(policyName, phase, version, modifiedDate);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        PhaseExecutionInfo other = (PhaseExecutionInfo) obj;
+        return Objects.equals(policyName, other.policyName) &&
+            Objects.equals(phase, other.phase) &&
+            Objects.equals(version, other.version) &&
+            Objects.equals(modifiedDate, other.modifiedDate);
+    }
+
+    @Override
+    public String toString() {
+        return Strings.toString(this, true, true);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(POLICY_NAME_FIELD.getPreferredName(), policyName);
+        builder.field(PHASE_DEFINITION_FIELD.getPreferredName(), phase);
+        builder.field(VERSION_FIELD.getPreferredName(), version);
+        if (builder.humanReadable()) {
```

Comment:
For human readable-ness, an additional field should be added, we shouldn't replace the field with a human readable version.

You should be able to do

```java
builder.timeField("modified_millis", "modified", modifiedDate);
```

(replacing the field names with the fields we want to use)

and then you don't have to check the human readable flag yourself